### PR TITLE
Fuzz fix timeout

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/resolve/resolved-dns-answer.c [hunk 1]
```diff
@@ -8,18 +8,52 @@
...
 
         if (n > UINT16_MAX) /* We can only place 64K RRs in an answer at max */
                 n = UINT16_MAX;
 
+        s = set_new(&dns_answer_item_hash_ops);
+        if (!s)
+                return NULL;
+
+        /* Higher multipliers give slightly higher efficiency through hash collisions, but the gains
+         * quickly drop of after 2. */
```
Comment (bluca) [inline src/resolve/resolved-dns-answer.c]: typo: /drop of/drop off/

File: src/resolve/resolved-dns-answer.c [hunk 2]
```diff
@@ -8,18 +8,52 @@
 #include "resolved-dns-dnssec.h"
 #include "string-util.h"
 
+static void dns_answer_item_hash_func(const DnsAnswerItem *a, struct siphash *state) {
+        assert(a);
```
Comment (bluca) [inline src/resolve/resolved-dns-answer.c:12]: should there be an assert on state too?

File: src/resolve/resolved-dns-answer.c [hunk 3]
```diff
@@ -54,12 +92,20 @@ static int dns_answer_add_raw(DnsAnswer *a, DnsResourceRecord *rr, int ifindex,
...
                 .rr = dns_resource_record_ref(rr),
                 .ifindex = ifindex,
                 .flags = flags,
         };
 
+        r = set_put(a->set_items, &a->items[a->n_rrs]);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return -EEXIST;
```
Comment (bluca) [inline src/resolve/resolved-dns-answer.c:106]: a->items[a->n_rrs].rr has been ref'ed above, what happens to it if we return early?

**Comment by bluca** [comment]
fuzzer CI is not happy:

```
    #0 0x7f183c2a6438 in raise (/lib/x86_64-linux-gnu/libc.so.6+0x35438)
    #1 0x7f183c2a8039 in abort (/lib/x86_64-linux-gnu/libc.so.6+0x37039)
    #2 0x7f183d6ed51f in log_assert_failed_realm /work/build/../../src/systemd/src/basic/log.c:863:9
    #3 0x7f183d72de44 in siphash24_compress /work/build/../../src/systemd/src/basic/siphash24.c:80:9
    #4 0x4e6d5c in dns_resource_record_hash_func /work/build/../../src/systemd/src/resolve/resolved-dns-rr.c
    #5 0x4f0dcc in dns_answer_item_hash_func /work/build/../../src/systemd/src/resolve/resolved-dns-answer.c:16:9
    #6 0x7f183d6d2a2f in base_bucket_hash /work/build/../../src/systemd/src/basic/hashmap.c:341:9
    #7 0x7f183d6d3974 in _hashmap_get /work/build/../../src/systemd/src/basic/hashmap.c:1340:16
    #8 0x4ea850 in set_get /work/build/../../src/systemd/src/basic/set.h:38:16
    #9 0x4ea5c0 in dns_answer_add /work/build/../../src/systemd/src/resolve/resolved-dns-answer.c:141:17
    #10 0x4da2b3 in dns_packet_extract_answer /work/build/../../src/systemd/src/resolve/resolved-dns-packet.c:2287:29
    #11 0x4d9561 in dns_packet_extract /work/build/../../src/systemd/src/resolve/resolved-dns-packet.c:2323:13
    #12 0x4cd991 in LLVMFuzzerTestOneInput /work/build/../../src/systemd/src/fuzz/fuzz-dns-packet.c:22:16
    #13 0x5423c6 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/libfuzzer/FuzzerLoop.cpp:599:15
    #14 0x540371 in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool, bool*) /src/libfuzzer/FuzzerLoop.cpp:505:3
    #15 0x545271 in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::__1::vector<fuzzer::SizedFile, fuzzer::fuzzer_allocator<fuzzer::SizedFile> >&) /src/libfuzzer/FuzzerLoop.cpp:814:7
    #16 0x545bf5 in fuzzer::Fuzzer::Loop(std::__1::vector<fuzzer::SizedFile, fuzzer::fuzzer_allocator<fuzzer::SizedFile> >&) /src/libfuzzer/FuzzerLoop.cpp:845:3
    #17 0x504a77 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/libfuzzer/FuzzerDriver.cpp:906:6
    #18 0x4f5fe7 in main /src/libfuzzer/FuzzerMain.cpp:20:10
    #19 0x7f183c29183f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2083f)
```

**Comment by yuwata** [comment]
@bluca Thank you for your review. The above points are addressed. PTAL.

**Review by bluca — APPROVED** [review]
LGTM!

**Comment by bluca** [comment]
Tests are not happy:

```
Assertion 'in' failed at src/basic/siphash24.c:80, function siphash24_compress(). Aborting.
```

**Comment by yuwata** [comment]
I found a mistake, and fixed it. Setting green label.

**Pull request merged**